x/evm/keeper/msg_server.go

```diff
@@ -347,6 +417,36 @@ func (server msgServer) AssociateContractAddress(goCtx context.Context, msg *typ
 	return &types.MsgAssociateContractAddressResponse{}, nil
 }
 
+func getEthReceipt(ctx sdk.Context, tx *ethtypes.Transaction, msg *core.Message, res *core.ExecutionResult, stateDB *state.DBImpl) *ethtypes.Receipt {
```

Comment:
this duplicates logic in `writeReceipt`. If this happens before `writeReceipt`, can you make it so that the result is passed over to `writeReceipt` so that we don't allocate memory/calculate bloom/etc. twice?

Reply:
Yeah indeed, I wasn't sure what you guys would have preferred. 

I'll check into refactoring this to share the computing.